app/core/BackgroundBridge/BackgroundBridge.js

```diff
@@ -557,6 +646,246 @@ export class BackgroundBridge extends EventEmitter {
+            ...params,
+            origin,
+          }),
+        getNonEvmAccountAddresses: Engine.controllerMessenger.call.bind(
+          Engine.controllerMessenger,
+          'MultichainRouter:getSupportedAccounts',
+        ),
+        trackSessionCreatedEvent: () => undefined,
+      }),
+    );
+
+    engine.push(
+      createUnsupportedMethodMiddleware(
+        new Set([
+          ...UNSUPPORTED_RPC_METHODS,
+          'eth_requestAccounts',
+          'eth_accounts',
+        ]),
+      ),
+    );
+
+    engine.push(
+      createMultichainMethodMiddleware({
+        // getProviderState handler related
+        getProviderState: this.getProviderState.bind(this),
+      }),
+    );
+
+    try {
+      const caip25Caveat = PermissionController.getCaveat(
+        origin,
+        Caip25EndowmentPermissionName,
+        Caip25CaveatType,
+      );
+
+      // add new notification subscriptions for changed authorizations
+      const sessionScopes = getSessionScopes(caip25Caveat.value, {
+        getNonEvmSupportedMethods: this.getNonEvmSupportedMethods.bind(this),
+      });
+
+      // if the eth_subscription notification is in the scope and eth_subscribe is in the methods
+      // then get the subscriptionManager going for that scope
+      Object.entries(sessionScopes).forEach(([scope, scopeObject]) => {
+        if (
+          scopeObject.notifications.includes('eth_subscription') &&
+          scopeObject.methods.includes('eth_subscribe')
+        ) {
+          this.addMultichainApiEthSubscriptionMiddleware({
+            scope,
+            origin,
+          });
+        }
+      });
+    } catch (err) {
+      // noop
+    }
+
+    this.multichainSubscriptionManager.on(
+      'notification',
+      (targetOrigin, message) => {
+        if (origin === targetOrigin) {
+          engine.emit('notification', message);
+        }
+      },
+    );
+
+    engine.push(
+      this.multichainMiddlewareManager.generateMultichainMiddlewareForOriginAndTabId(
+        origin,
+      ),
+    );
+
+    // user-facing RPC methods
+    engine.push(
+      this.createMiddleware({
+        hostname: this.hostname,
+        getProviderState: this.getProviderState.bind(this),
+      }),
+    );
+
+    engine.push(async (req, res, _next, end) => {
+      const { provider } = NetworkController.getNetworkClientById(
+        req.networkClientId,
+      );
+      res.result = await provider.request(req);
+      return end();
+    });
+
+    return engine;
+  }
+
+  /**
+   * This handles CAIP-25 authorization changes every time relevant permission state changes, for any reason.
+   */
+  setupCaipEventSubscriptions() {
+    const { controllerMessenger, context } = Engine;
+
+    // wallet_sessionChanged and eth_subscription setup/teardown
+    controllerMessenger.subscribe(
+      `${context.PermissionController.name}:stateChange`,
+      this.handleCaipSessionScopeChanges(),
```

Comment:
```suggestion
      this.handleCaipSessionScopeChanges,
```

maybe even `.bind(this)`